Bond/Bond+Arrays.swift

```diff
@@ -372,15 +388,15 @@ private class DynamicArrayFilterProxy<T>: DynamicArray<T> {
     self.filterf = filterf
     self.bond = ArrayBond<T>()
     self.bond.bind(sourceArray, fire: false)
-    
-    super.init([])
-    
+
```

Comment:
I had to move `super.init([])` after `pointers` is set up, so that the initial count that `DynamicArray` reads will be correct.